020-Getting-started/070-nextjs.mdx

```diff
@@ -50,185 +46,621 @@ apiKey=YOUR_API_KEY_HERE
-// We'll assign the client to xata to make it easier to call
 const xata = getXataClient();
 
-// We will make the query to Xata on the server.
-// This will allow the query to read the API_KEY from the .env file
-export const getServerSideProps = async () => {
-  const records = await xata.db.Posts.getMany();
-  return {
-    props: {
-      records: records.map((record) => record.toSerializable())
-    }
-  };
-};
-
-// Pass the props defined above to the page and map against it
-export default function PostsListPage({ records }: InferGetServerSidePropsType<typeof getServerSideProps>) {
+export default async function Home({
+  searchParams
+}: {
+  searchParams: { q: string };
+}) {
+  let posts = null;
+  if (searchParams.q) {
+    posts = await xata.db.Posts.search(searchParams.q, { fuzziness: 2 });
+  } else {
+    posts = await xata.db.Posts.getAll();
+  }
+
   return (
-    <>
-      <h1>My xata posts</h1>
-      {records.map((record) => (
-        <div key={record.id}>
-          <h2>{record.title}</h2>
-          <p>{record.id}</p>
-          <p>{record.body}</p>
-          <p>{record.pubDate}</p>
+    <main className="flex flex-wrap justify-center content-start p-8 lg:p-24 min-h-screen">
+      <div className="z-10 h-50 w-full max-w-5xl items-center justify-between text-xl lg:flex">
+        <p className="fixed left-0 top-0 flex w-full justify-center pb-6 pt-8 lg:static lg:w-auto bg-gradient-to-b from-white via-white via-65% dark:from-black dark:via-black lg:bg-none">
+          <a href="/">Get started with Xata and Next.js</a>
+        </p>
+        <div className="fixed bottom-0 left-0 flex w-full items-end justify-center bg-gradient-to-t from-white via-white dark:from-black dark:via-black lg:static lg:h-auto lg:w-auto lg:bg-none">
+          <a href="https://xata.io" className="w-20">
+            <img src="https://raw.githubusercontent.com/leggetter/xata-examples/5b0715c9ba36556400b6121bd224203cb89f7741/docs/app_logo.svg" />
+          </a>
         </div>
-      ))}
-    </>
+      </div>
+      <div className="w-full max-w-5xl mt-16">
+        <form>
+          <input
+            name="q"
+            defaultValue={searchParams.q}
+            placeholder="Search..."
+            className="w-full rounded-lg p-2 dark:text-purple-950"
+          />
+        </form>
+      </div>
+      ...
+    </main>
   );
 }
 ```
 
-In Next.js, it is recommended to use the `InferGetServerSidePropsType` helper to infer the type of the props returned by `getServerSideProps`, so that types match the query results.
+Add search with three changes.
+
+First, update the main function to accept search parameters. This is achieved by having the function accept an object
+with a `searchParams` property. When a search has been performed, `searchParams` will optionally have a `q` property
+access via `searchParams.q`.
+
+```tsx
+export default async function Home({ searchParams }: { searchParams: { q: string } }) {
```

Comment:
Nit: This was a bit confusing because it is formatted differently than in the code snippet above.

Reply:
Good spot. I had auto-formatting turned on for a while and must have missed this change. Thanks!